src/methods/correlation.ts

```diff
@@ -1,15 +1,22 @@
-import { SimpleDataItem, Options, defaultOptions } from "../types/SimpleData.types.js"
+import { SimpleDataItem } from "../types/SimpleData.types.js"
 import { sampleCorrelation, combinations } from "simple-statistics"
 import checkTypeOfKey from "../helpers/checkTypeOfKey.js"
 import hasKey from "../helpers/hasKey.js"
 
-export default function correlation(data: SimpleDataItem[], key1: string, key2: string, options: Options): any[] {
+export default function correlation(
+    data: SimpleDataItem[], 
+    verbose: boolean, 
+    numDigits: number, 
+    nbTestedValues: number,
+    key1?: string, 
+    key2?: string
+): any[] {
 
     const correlations = []
 
     if (key1 === undefined && key2 === undefined) {
 
-        const keys = Object.keys(data[0]).filter(d => checkTypeOfKey(data, d, "number", 1, options))
+        const keys = Object.keys(data[0]).filter(d => typeof d === "number")
```

Comment:
I don't think this works. You are filtering the keys, which are all strings. You want to filter the values. I would keep the checkTypeOfKey function here.